portal/code/company-portal/src/components/cax-header.tsx

```diff
@@ -71,16 +62,16 @@ class Header extends React.Component<IProp> {
                 </Col>
                 <Col>
                     <div className='d-flex flex-row-reverse profile'>
-                        <div className='profile-lang'><span className={this.language === 'en' ? 'lang-sel' : ''} onClick={() => changeLanguage('en')} > EN </span></div>
-                        <div className='profile-lang'><span className={this.language === 'de' ? 'lang-sel' : ''} onClick={() => changeLanguage('de')} > DE</span></div>
+                        <div className='profile-lang'><span className={language === 'en' ? 'lang-sel' : ''} onClick={() => changeLanguage('en')} > EN </span></div>
+                        <div className='profile-lang'><span className={language === 'de' ? 'lang-sel' : ''} onClick={() => changeLanguage('de')} > DE</span></div>
                         <div className='profile-link partion'></div>
                         <div className='profile-link user'>
                             <input id="myid" type="checkbox" />
                             <label htmlFor="myid" className='user-icon'></label>
-                            <span className="tooltiptext"> <div> {this.username}</div>
+                            <span className="tooltiptext"> <div> {username}</div>
                                 < div > {UserService.getDomain()}</div>
-                                <div>({this.userRoles.join(", ")})</div>
-                                <div className='logout' onClick={() => this.logoutClick()}>Logout</div>
+                                <div>({userRoles.join(", ")})</div>
+                                <div className='logout' onClick={() => logoutClick()}>Logout</div>
```

Comment:
```suggestion
                                <div className='logout' onClick={() => UserService.doLogout()}>Logout</div>
```

Reply:
although "Logout" is probably understood by everyone - shouldn't this string be i18n

Reply:
updated the changes